src/CommunityToolkit/Xamarin.CommunityToolkit/Behaviors/EventToCommandBehaviorGeneric.shared.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+
+namespace Xamarin.CommunityToolkit.Behaviors
+{
+	/// <summary>
+	/// This <see cref="EventToCommandBehavior"/> cast the sender object to a specific type defined by the user.
+	/// </summary>
+	/// <typeparam name="TType">The type that you want to receive in your <see cref="Xamarin.Forms.Command"/> </typeparam>
+	public sealed class EventToCommandBehavior<TType> : EventToCommandBehavior
+	{
+		protected override void OnTriggerHandled(object? sender = null, object? eventArgs = null)
+		{
+			var parameter = CommandParameter
+				?? EventArgsConverter?.Convert(eventArgs, typeof(object), null, null)
+				?? eventArgs;
+
+			if (parameter is not TType)
+			{
+				// changing it to the default value to avoid a cast exception
+				parameter = default(TType);
+				parameter = Convert.ChangeType(parameter, typeof(TType));
```

Comment:
Can you please clarify why do we need to call ```Convert.ChangeType(parameter, typeof(TType))``` if parameter is already typeof TType here (cuz we set it in the line above)

Reply:
I believe that this isn't needed anymore, I can't remember why I did it in first place, I'll try without it and see how it goes, if works I push a new version.

Reply:
Also, please consider my suggestion below)